### DISCUSSION TITLE
Recommeded GPU card for LBM simulations

### DISCUSSION THREAD

**Discussion by rdapena** [espressomd/espresso#4926] [Q&A] [answered]
Hi!

I've recently started using espresso for LBM simulations to study the interaction between RBCs and particles with different shapes. The research group I'm part of can acquire a new GPU specifically tailored to run such simulations, but as we are just starting out, we don't know what specifications it should have and what architecture would be better suited. If anyone here has experience in this area I'd appreciate the help.

Thanks

**Comment by jngrad** [comment, marked as answer]
Our team regularly uses [NVIDIA A100 PCIe](https://www.nvidia.com/content/dam/en-zz/Solutions/Data-Center/a100/pdf/nvidia-a100-datasheet-nvidia-us-2188504-web.pdf) (Ampere, 64 Gbps, 40 GB HBM2e) and [NVIDIA L4](https://www.nvidia.com/en-us/data-center/l4/) (Lovelace, 24 GB GDDR6) on clusters, and a mix of GeForce RTX 4070 (Lovelace), GeForce RTX 3070 Ti (Ampere) and GeForce RTX 2080 (Turing) on desktop workstations. Unfortunately we don't have precise benchmarks for these different GPUs, because we have only just recently completed the LB GPU code rewrite on the main branch, and are still working out a few performance improvements. You will probably want to use the 4.2.2 release of ESPResSo, where particle coupling on the GPU is 20 to 40 times faster than on the CPU on a desktop machine.

Here are a few hints:

* the particle coupling will be the performance bottleneck, especially if you use multiple CPUs for one GPU
* the GPU VRAM plays a role if you intend to simulate very large LB grids, for example a NVIDIA L4 VRAM is already maxed out with a 256x256x256 LB grid in single-precision floating point format
* ESPResSo will require CUDA SDK 12.1 later this year, so you may prefer a GPU that will support it ([compatibility matrix](https://en.wikipedia.org/wiki/CUDA#GPUs_supported))

**Reply by rdapena** [reply]
Thank you for the detailed reply. It's been extremely helpful.
